strawberry/field.py

```diff
@@ -134,8 +135,22 @@ def __init__(
         self.is_subscription = is_subscription
 
         self.permission_classes: List[Type[BasePermission]] = list(permission_classes)
-        self.directives = directives
-
+        self.directives = list(directives)
+        self.extensions: Optional[List[FieldExtension]] = extensions
+
+        # Automatically add the permissions extension
+        if len(self.permission_classes):
```

Comment:
Are we going to deprecate the `permission_classes` usage and have only `extensions`? If so, it would be nice to add a deprecation warning here

Reply:
Between:

`@strawberry.field(extensions=[PermissionExtension(permissions=[IsAuthorized()])])`

and

`@strawberry.field(permission_classes=[IsAuthorized()])`

I still prefer the latter, it's less code to write. What do you think of only using `PermissionExtension` internally and keeping the same api we have now? :)

Reply:
There's a third option:

```python
@strawberry.field(extensions=[IsAuthorized()])
```

And have the extensions wrap the permission, just like it does when passing them as `permission_classes`.

Or even change the permissions to be a subclass of `FieldExtension`, this way permissions would basically be the same as field extensions without any special handling